pkg/image/prune/prune.go

```diff
@@ -1068,14 +1066,16 @@ func deleteFromRegistry(registryClient *http.Client, url string) error {
 	deleteFunc := func(proto, url string) error {
 		req, err := http.NewRequest("DELETE", url, nil)
 		if err != nil {
-			glog.Errorf("Error creating request: %v", err)
-			return fmt.Errorf("error creating request: %v", err)
+			return err
 		}
 
 		glog.V(4).Infof("Sending request to registry")
 		resp, err := registryClient.Do(req)
 		if err != nil {
-			return fmt.Errorf("error sending request: %v", err)
+			if proto != "https" && strings.Contains(err.Error(), "malformed HTTP response") {
+				return fmt.Errorf("%v.\n* Are you trying to connect to a TLS-enabled registry without TLS?", err)
+			}
```

Comment:
:+1: for the hint. Nevertheless, this should have been captured by the `ping()` call.

I must have forgotten to modify this code same the way as the `ping()` (see https://github.com/openshift/origin/blob/90592c5815e371f648705c8dab384e36f21e8457/pkg/image/prune/prune.go#L199) - to exclude HTTP connection from attempts when secure connection is requested. If you want, you can amend it here, otherwise I'll do it after this lands.